commands/update_party_profile.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2023 Gobalsky Labs Limited
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package commands
+
+import (
+	"fmt"
+
+	vgstrings "code.vegaprotocol.io/vega/libs/strings"
+	commandspb "code.vegaprotocol.io/vega/protos/vega/commands/v1"
+)
+
+func CheckUpdatePartyProfile(cmd *commandspb.UpdatePartyProfile) error {
+	return checkUpdatePartyProfile(cmd).ErrorOrNil()
+}
+
+func checkUpdatePartyProfile(cmd *commandspb.UpdatePartyProfile) Errors {
+	errs := NewErrors()
+
+	if cmd == nil {
+		return errs.FinalAddForProperty("update_party_profile", ErrIsRequired)
+	}
+
+	if len(cmd.Alias) > 32 {
+		errs.AddForProperty("update_party_profile.alias", ErrIsLimitedTo32Characters)
+	}
+
+	cmd.Alias = vgstrings.TrimBlank(cmd.Alias)
+
+	if len(cmd.Metadata) > 10 {
+		errs.AddForProperty("update_party_profile.metadata", ErrIsLimitedTo10Entries)
+	} else {
+		for i, m := range cmd.Metadata {
+			if len(m.Key) > 32 {
```

Comment:
Do want want to check that they don't have two things in the meta-data slice with the same key? Because once it gets into core I guess whichever was first would just disappear.

Reply:
I can do it if we think it's important.

Reply:
It is seen as necessary, so I will do it